tests/unit/OptionsListUtilsTest.ts

```diff
@@ -983,6 +983,16 @@ describe('OptionsListUtils', () => {
                     expect(filteredResults.recentReports.at(0)?.text).toBe('The Flash');
                 });
         });
+
+        it('should return all results when setting maxRecentReportsToShow to 0', () => {
+            const options = OptionsListUtils.getSearchOptions(OPTIONS);
+            const filteredOptions = OptionsListUtils.filterAndOrderOptions(options, '', {maxRecentReportsToShow: 0});
+
+            expect(filteredOptions.recentReports.length).toBe(10);
```

Comment:
This doesn't make sense to me. We're asking it show 0 recent reports max, then we assert it shows 10. I think we should leave the value undefined if we don't want to filter at all. If it's 0, we should return 0 recent reports.

Reply:
yeah i get why that one is confusing, let me explain:

- before those refactorings `maxRecentReportsToShow` was by default `0`. When passing 0 it was expected to return all reports. 
- as there were more call sites passing `maxRecentReportsToShow: 0` then actually limiting them, i changed the parameter to be optional and by default return all reports. I removed all call sites using `maxRecentReportsToShow: 0`
- [In this review comment](https://github.com/Expensify/App/pull/53371#discussion_r1868351909) @ahmedGaber93 pointed out that we should add a test to verify that `maxRecentReportsToShow: 0` are still returning all reports and i added this test

Now i realized and agree with Neil that passing `0` is confusing and you'd expect to get `0` elements back in that case (as the parameter is optional now and the default is to return all items).

I extended one test case that now verifies that when passing `maxRecentReportsToShow: 0` no report results are returned here: https://github.com/Expensify/App/pull/53371/commits/dfa6fd6347d945c4a1190d3f9001901360617ab3

I hope this is fine now with everyone?